src/Store/MongoDB/Store.php

```diff
@@ -0,0 +1,133 @@
+ *
+ * @author Oskar Stark <oskarstark@googlemail.com>
+ */
+final readonly class Store implements VectorStoreInterface
+{
+    /**
+     * @param string $databaseName    The name of the database
+     * @param string $collectionName  The name of the collection
+     * @param string $indexName       The name of the index
+     * @param string $vectorFieldName The name of the field int the index that contains the vector
+     */
+    public function __construct(
+        private Client $client,
+        private LoggerInterface $logger,
+        private string $databaseName,
+        private string $collectionName,
+        private string $indexName,
+        private string $vectorFieldName = 'vector',
+    ) {
+    }
+
+    public function addDocument(Document $document): void
+    {
+        $this->addDocuments([$document]);
+    }
+
+    public function addDocuments(array $documents): void
+    {
+        $collection = $this->client->selectCollection($this->databaseName, $this->collectionName);
+
+        foreach ($documents as $document) {
+            if (!$document->hasVector()) {
+                $this->logger->warning('Document {id} does not have a vector', ['id' => $document->id]);
+            }
+
+            $id = $document->id->toRfc4122();
+
+            $collection->replaceOne(
+                ['id' => $id],
+                array_filter([
+                    'metadata' => $document->metadata->getArrayCopy(),
+                    $this->vectorFieldName => $document->vector->getData(),
+                    'id' => $id,
+                    'text' => $document->text,
+                ]),
+                ['upsert' => true], // insert if not exists
+            );
+        }
+    }
+
+    /**
+     * @param array{
+     *     limit?: positive-int,
+     *     numCandidates?: positive-int,
+     * } $options
+     *
+     * @return array|Document[]
+     */
+    public function query(Vector $vector, array $options = []): array
+    {
+        $collection = $this->client->selectCollection($this->databaseName, $this->collectionName);
+
+        $results = $collection->aggregate([
+            [
+                '$vectorSearch' => [
+                    'index' => $this->indexName,
+                    'path' => $this->vectorFieldName,
+                    'queryVector' => $vector->getData(),
+                    'numCandidates' => $options['numCandidates'] ?? 200,
+                    'limit' => $options['limit'] ?? 5,
+                ],
+            ],
+            [
+                '$project' => [
+                    '_id' => 0,
+                    'id' => 1,
+                    'text' => 1,
+                    'metadata' => 1,
+                    $this->vectorFieldName => 1,
+                    'score' => [
+                        '$meta' => 'vectorSearchScore',
+                    ],
+                ],
+            ],
```

Comment:
Unless you want to specifically exclude other fields than ID, I recommend not using `$project` but rather `$addFields`. Assuming you're using the default `_id` field for the identifier, the following stage would accomplish what you want:
```suggestion
            [
                '$addFields' => [
                    'id' => '$_id',
                    'score' => [ '$meta' => 'vectorSearchScore' ],
                ],
            ],
```